### PR TITLE
Assorted fixes and improvements

### PR DESCRIPTION
**Pull request by pkryger**
I decided to submit this as a single PR, as majority of these commits are limited in scope to a few lines.

These are various fixes (like no `-i` for `zsh`, or late `desktop` load) as well as some improvements. In the past I had quite a good track record with CI stability on this branch, so I guess some of them are fixes indeed (the single packages thingy perhaps?).

### CODE REVIEW

File: modules/init-cpp.el [hunk 1]
```diff
@@ -23,8 +23,7 @@
 (use-package iedit
   :init
   ;;; Fix A bug (normal key is "C-;")
-  :bind (:map global-map
-              ("C-c ;" . #'iedit-mode)))
+  :bind ("C-c ;" . iedit-mode))
```
Comment (philippe-grenet) [inline modules/init-cpp.el:26]: Are you removing all the `#'` for consistency throughout the code?

File: modules/init-helm.el [hunk 2]
```diff
@@ -72,6 +72,8 @@
   (("C-S-r" . helm-rg)))
 
 (use-package helm-swoop
+  :custom
+  (helm-swoop-split-direction 'split-window-horizontally)
```
Comment (philippe-grenet) [inline modules/init-helm.el:76]: Hmmm that one isn't necessarily my preference 😃 . But it makes sense to make it opinionated. I can override this in my taps, so no issue there.

**Review by philippe-grenet — APPROVED** [review]
LGTM, I only left remarks or questions. The only thing is you should consider adding the new key-bindings in the README (those added in "Normalize keybindings" and the projectile-related commits)

File: modules/init-cpp.el (hunk 1 above)
Comment (pkryger) [reply modules/init-cpp.el:26]: indeed

Comment (steve-downey) [reply modules/init-cpp.el:26]: use-package takes care of it for us, it's only really needed in a raw key binding where we to wrap the call we want. See
https://www.gnu.org/software/emacs/manual/html_node/elisp/Anonymous-Functions.html#index-_0023_0027-syntax

**Pull request merged**